### ISSUE TITLE
Call to undefined method OCA\Deck\Event\FTSEvent::hasArgument()

### ISSUE THREAD

**Issue by paulb-opusvl** [nextcloud/deck#2360]
**Describe the bug**
When creating, updating or deleting a card the UI produces odd behaviour in that it doesn't save or will save an create multiple cards.

Using F12 Dev tools I see I get the response to a DELETE:

```json
{
  "status": 500,
  "message": "Call to undefined method OCA\\Deck\\Event\\FTSEvent::hasArgument()",
  "exception": {
    "\u0000*\u0000message": "Call to undefined method OCA\\Deck\\Event\\FTSEvent::hasArgument()",
    "\u0000Exception\u0000string": "",
    "\u0000*\u0000code": 0,
    "\u0000*\u0000file": "\/var\/www\/html\/lib\/private\/AppFramework\/Http\/Dispatcher.php",
    "\u0000*\u0000line": 110,
    "\u0000Exception\u0000trace": [
      {
        "file": "\/var\/www\/html\/lib\/private\/AppFramework\/App.php",
        "line": 137,
        "function": "dispatch",
        "class": "OC\\AppFramework\\Http\\Dispatcher",
        "type": "-\u003E",
        "args": [
          {},
          "delete"
        ]
      },
      {
        "file": "\/var\/www\/html\/lib\/private\/AppFramework\/Routing\/RouteActionHandler.php",
        "line": 47,
        "function": "main",
        "class": "OC\\AppFramework\\App",
        "type": "::",
        "args": [
          "OCA\\Deck\\Controller\\CardController",
          "delete",
          {},
          {
            "cardId": "61",
            "_route": "deck.card.delete"
          }
        ]
      },
      {
        "function": "__invoke",
        "class": "OC\\AppFramework\\Routing\\RouteActionHandler",
        "type": "-\u003E",
        "args": [
          {
            "cardId": "61",
            "_route": "deck.card.delete"
          }
        ]
      },
      {
        "file": "\/var\/www\/html\/lib\/private\/Route\/Router.php",
        "line": 297,
        "function": "call_user_func",
        "args": [
          {},
          {
            "cardId": "61",
            "_route": "deck.card.delete"
          }
        ]
      },
      {
        "file": "\/var\/www\/html\/lib\/base.php",
        "line": 1012,
        "function": "match",
        "class": "OC\\Route\\Router",
        "type": "-\u003E",
        "args": [
          "\/apps\/deck\/cards\/61"
        ]
      },
      {
        "file": "\/var\/www\/html\/index.php",
        "line": 37,
        "function": "handleRequest",
        "class": "OC",
        "type": "::",
        "args": []
      }
    ],
    "\u0000Exception\u0000previous": {}
  }
}
```

**To Reproduce**
1. Go to any card in the deck
2. Click on the ... to drop the menu and choose delete
3. See card is not deleted
4. View F12 dev tools and see the server response above

**Expected behavior**
A card should be deleted

**Client details:**
 - OS: Debian Buster Linux
 - Browser Vivaldi / Chrome / Firefox
 - Version 
 - Device: Desktop

<details>
<summary>Server details</summary>

**Operating system**: Debian Buster Linux

**Web server:** Docker image nextcloud:19.0.3

**Database:**

**PHP version:**

**Nextcloud version:** 19.0.3

**Where did you install Nextcloud from:** Docker

**Signing status:**

```
Login as admin user into your Nextcloud and access
http://example.com/index.php/settings/integrity/failed
paste the results here.
```

**List of activated apps:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```

```text
Enabled:
  - accessibility: 1.5.0
  - activity: 2.12.0
  - calendar: 2.0.4
  - cloud_federation_api: 1.2.0
  - comments: 1.9.0
  - contactsinteraction: 1.0.0
  - dav: 1.15.0
  - deck: 1.1.0
  - duplicatefinder: 0.0.2
  - encryption: 2.7.0
  - federatedfilesharing: 1.9.0
  - federation: 1.9.0
  - files: 1.14.0
  - files_mindmap: 0.0.22
  - files_pdfviewer: 1.8.0
  - files_rightclick: 0.16.0
  - files_sharing: 1.11.0
  - files_trashbin: 1.9.0
  - files_versions: 1.12.0
  - files_videoplayer: 1.8.0
  - firstrunwizard: 2.8.0
  - forms: 2.0.4
  - fulltextsearch: 1.4.2
  - groupfolders: 7.0.0
  - ldapcontacts: 2.0.3
  - logreader: 2.4.0
  - lookup_server_connector: 1.7.0
  - mail: 1.4.1
  - nextcloud_announcements: 1.8.0
  - notifications: 2.7.0
  - oauth2: 1.7.0
  - ojsxc: 4.1.1
  - password_policy: 1.9.1
  - photos: 1.1.0
  - polls: 1.4.3
  - printer: 0.0.3
  - privacy: 1.3.0
  - provisioning_api: 1.9.0
  - ransomware_protection: 1.7.0
  - recommendations: 0.7.0
  - serverinfo: 1.9.0
  - settings: 1.1.0
  - sharebymail: 1.9.0
  - support: 1.2.1
  - survey_client: 1.7.0
  - suspicious_login: 3.2.1
  - systemtags: 1.9.0
  - text: 3.0.1
  - theming: 1.10.0
  - twofactor_backupcodes: 1.8.0
  - updatenotification: 1.9.0
  - user_ldap: 1.9.0
  - viewer: 1.3.0
  - workflow_script: 1.4.0
  - workflowengine: 2.1.0
Disabled:
  - admin_audit
  - files_external
  - grauphel
  - preferred_providers
  - whiteboard
```

**Nextcloud configuration:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or

Insert your config.php content here
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
$CONFIG = array (
  'htaccess.RewriteBase' => '/',
  'memcache.local' => '\\OC\\Memcache\\APCu',
  'apps_paths' => 
  array (
    0 => 
    array (
      'path' => '/var/www/html/apps',
      'url' => '/apps',
      'writable' => false,
    ),
    1 => 
    array (
      'path' => '/var/www/html/custom_apps',
      'url' => '/custom_apps',
      'writable' => true,
    ),
  ),
  'memcache.distributed' => '\\OC\\Memcache\\Redis',
  'memcache.locking' => '\\OC\\Memcache\\Redis',
  'redis' => 
  array (
    'host' => 'redis',
    'password' => false,
    'port' => 6379,
  ),
  'passwordsalt' => 'xxx',
  'secret' => 'xxx',
  'trusted_domains' => 
  array (
    0 => 'share.opusvl.io',
  ),
  'trusted_proxies' => 
  array (
    0 => '172.24.0.1',
  ),
  'datadirectory' => '/var/www/html/data',
  'dbtype' => 'pgsql',
  'version' => '19.0.3.1',
  'overwrite.cli.url' => 'https://share.opusvl.io',
  'overwriteprotocol' => 'https',
  'dbname' => 'nextcloud',
  'dbhost' => 'postgresql',
  'dbport' => '',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'oc_admin',
  'dbpassword' => 'xxx',
  'installed' => true,
  'instanceid' => 'xxx',
  'loglevel' => 2,
  'maintenance' => false,
  'mail_smtpmode' => 'smtp',
  'mail_smtphost' => 'msa.mail.opusvl.com',
  'mail_sendmailmode' => 'smtp',
  'mail_smtpport' => '587',
  'updater.secret' => 'xxx',
  'ldapProviderFactory' => 'OCA\\User_LDAP\\LDAPProviderFactory',
);
```

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

</details>

<details>
<summary>Logs</summary>

#### Nextcloud log (data/nextcloud.log)
```
nextcloud_1   | 172.29.0.1 - - [06/Oct/2020:09:56:08 +0000] "DELETE /apps/deck/cards/62 HTTP/1.0" 500 2194 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.123 Safari/537.36"
```

#### Browser log
```
VM1274:1 DELETE https://share.opusvl.io/apps/deck/cards/62 500
(anonymous) @ VM1274:1
(anonymous) @ xhr.js:184
e.exports @ xhr.js:13
e.exports @ dispatchRequest.js:50
Promise.then (async)
c.request @ Axios.js:61
r.forEach.c.<computed> @ Axios.js:76
(anonymous) @ bind.js:9
value @ CardApi.js:49
(anonymous) @ card.js:202
c @ runtime.js:45
(anonymous) @ runtime.js:274
forEach.e.<computed> @ runtime.js:97
w @ deck.js?v=b36931d5-10:978
i @ deck.js?v=b36931d5-10:978
(anonymous) @ deck.js?v=b36931d5-10:978
(anonymous) @ deck.js?v=b36931d5-10:978
deleteCard @ card.js:201
(anonymous) @ vuex.esm.js:847
d.dispatch @ vuex.esm.js:512
dispatch @ vuex.esm.js:402
deleteCard @ CardMenu.vue:131
click @ CardMenu.vue?e6df:1
He @ vue.runtime.esm.js:1854
n @ vue.runtime.esm.js:2179
He @ vue.runtime.esm.js:1854
e.$emit @ vue.runtime.esm.js:3888
onClick @ ncvuecomponents.js:66
He @ vue.runtime.esm.js:1854
n @ vue.runtime.esm.js:2179
a._wrapper @ vue.runtime.esm.js:6917
createError.js:16 Uncaught (in promise) Error: Request failed with status code 500
    at e.exports (createError.js:16)
    at e.exports (settle.js:17)
    at XMLHttpRequest.p.onreadystatechange (xhr.js:69)
e.exports @ createError.js:16
e.exports @ settle.js:17
p.onreadystatechange @ xhr.js:69
Promise.then (async)
(anonymous) @ vuex.esm.js:515
d.dispatch @ vuex.esm.js:514
dispatch @ vuex.esm.js:402
deleteCard @ CardMenu.vue:131
click @ CardMenu.vue?e6df:1
He @ vue.runtime.esm.js:1854
n @ vue.runtime.esm.js:2179
He @ vue.runtime.esm.js:1854
e.$emit @ vue.runtime.esm.js:3888
onClick @ ncvuecomponents.js:66
He @ vue.runtime.esm.js:1854
n @ vue.runtime.esm.js:2179
a._wrapper @ vue.runtime.esm.js:6917
```

</details>

**Comment by paulb-opusvl** [comment]
Clearly something wrong with our implementation as this works in a fresh build.

**Comment by paulb-opusvl** [comment]
When I restored the database into the fresh build then the problem resurfaced. I'm at a loss as to what has happened here.

**Comment by juliushaertl** [comment]
Fix is pending in #2358
